.github/workflows/ci.yaml

```diff
@@ -7,18 +7,28 @@ on:
     branches:
       - main
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.event_name == 'pull_request' }}
+
```

Comment:
I like this because it prevents us from kicking jobs for different commits at the same time, but there's one thing about this I don't quite like: if two different jobs fail for the latest commit, you either have to retrigger them all or re-trigger one, wait for that to pass and re-trigger the second one.

I've looked it up and there's no way to tweak the concurrency per individual job, only per workflow. Any thoughts?